### DISCUSSION TITLE
Hiding database details?

### DISCUSSION THREAD

**Discussion by sharkAndshark** [maplibre/martin#429] [General]
Currently front end can know that it's a database function when they call /rpc/somemething/Z/X/y.pbf.
And I'm not sure that is the API a little verbose?

**Comment by nyurik** [comment]
Thanks @sharkAndshark , I'm [already working](https://github.com/maplibre/martin/compare/main...nyurik:martin:sources?expand=1) on that :)

Basic idea:  there will be "sources", each with its own ID, all mixable (i.e. you can ask for `/id1,id2,id3/z/x/y.mvt`), each source could come from a different backend (pg, mbtiles, ...), and it will support more than just mvt tiles (png/jpeg/...).

P.S. Note that this will most likely be a breaking change, at least for the configuration.  I might keep the `/rpc/...` for backwards compat as a simple redirect that won't even test if its rpc or not.

**Comment by nyurik** [comment]
I started a proper discussion on restructuring the endponts -- https://github.com/maplibre/martin/discussions/430 -- lets continue there
